### PR TITLE
add THIRD_PARTY_PATH flag

### PR DESCRIPTION
**Pull request by tianshuo78520a**
### PR types

Others
### PR changes

Others
### Describe

Add flag of xly mac ci

### CODE REVIEW

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -241,7 +242,8 @@ EOF
         -DPY_VERSION=${PY_VERSION:-2.7} \
         -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX:-/paddle/build} \
         -DWITH_GRPC=${grpc_flag} \
-        -DWITH_LITE=${WITH_LITE:-OFF};build_error=$?
+        -DWITH_LITE=${WITH_LITE:-OFF};build_error=$? \
+        -DTHIRD_PARTY_PATH=${THIRD_PARTY_PATH:-}
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]:
```
-DWITH_LITE=${WITH_LITE:-OFF} \
-DTHIRD_PARTY_PATH=${THIRD_PARTY_PATH:-};build_error=$? 
```

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -250,6 +252,9 @@ EOF
 function cmake_gen() {
     mkdir -p ${PADDLE_ROOT}/build
     cd ${PADDLE_ROOT}/build
+    if [ ! "${THIRD_PARTY_PATH}" ];then
+        THIRD_PARTY_PATH=${PADDLE_ROOT}/build
+    fi
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: 默认就是build/third_party目录，这里可以不用设置了
https://github.com/PaddlePaddle/Paddle/blob/d1047d0a699689724e037f944f205fbeca7ab1fb/cmake/third_party.cmake#L18-L20

Comment (tianshuo78520a) [reply paddle/scripts/paddle_build.sh]: 如果不输入会报错
image

File: paddle/scripts/paddle_build.sh (hunk 1 above)
Comment (tianshuo78520a) [reply paddle/scripts/paddle_build.sh]: 已修改

**Pull request closed without merging**